lib/puppet/network/http/connection.rb

```diff
@@ -313,14 +313,18 @@ def with_connection(site, &block)
       end
       response
     rescue OpenSSL::SSL::SSLError => error
+      self.class.handle_connection_error!(error, @verify, site.host)
+    end
+
+    def self.handle_connection_error!(error, verifier, host)
```

Comment:
I could be wrong, but I feel like this shouldn't have the exclamation point here. In Ruby that usually is used to indicate a destructive/mutating operation on a collection or object.